docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+#### Enable codegen in your package.json
+
+At this point, you are now ready to enable code-gen support in your library. In your library’s package.json add the following:
+
+```json
+// in package.json
+// Please note that this format is subject to change.
+"codegenConfig": {
+  "libraries": [
+    {
+      "name": "YourTurboModuleSpec",
+      "type": "modules",
+      "jsSrcsDir": "Libraries"
+    },
+    {
+      "name": "YourComponentName",
+      "type": "components",
+      "jsSrcsDir": "Libraries"
+    }
+  ]
+}
+```
+
+There's three arguments that are required:
+
+- `name`: A name of your library. This will be used to determine import path for your library.
+- `jsSrcsDir`: Path to the directory that contains the JavaScript specs for this library.
+
+These arguments are optional:
+
+- `type`: Optional. A string that determines which types of artifacts will be generated for your library: “modules” or “components”. If left unspecified, both modules and components artifacts will be generated.
+
+### 2. Extend or implement the code-generated native interfaces
+
+The JavaScript spec for your native module or component will be used to generate native interface code for each supported platform (i.e. Android and iOS). These native interface files will be generated when a React Native application that depends on your library is built.
+
+While this generated native interface code will not ship as part of your library, you do need to make sure your Objective-C or Java code conforms to the protocols provided by these native interface files. You can use the code-gen script to generate your library’s native interface code in order to use as reference. The files that are output by the script should not be committed, but you’ll need to refer to them to determine what changes you need to make to your native modules in order for them to provide an implementation for each generated @protocol / native interface.
+
+#### Conform to the protocols provided by the native interface code
+
+Update your native module or component to ensure it implements/extends the native interface that has been code-generated from your JavaScript specs.
+
+Following the example set forth in the previous section, your library might import MyAwesomeSpecs.h, extend the relevant native interface, and implement the necessary methods for this interface:
+
+```objc
+#import <MyAwesomeSpecs/MyAwesomeSpecs.h>
+
+@interface MyAwesomeModule () <StringGetterSpec>
+@end
+
+RCT_EXPORT_METHOD(getString:(NSString *)string
+                   callback:(RCTResponseSenderBlock)callback)
+{
+  // ...
+}
```

Comment:
I think it's also needed to add

```objc
- (std::shared_ptr<TurboModule>)getTurboModule:(const ObjCTurboModule::InitParams &)params
{
  return std::make_shared<StringGetterSpecJSI>(params);
}
```

Reply:
@sota000 can you confirm this?

Reply:
Hey Janic, I looked into the changes I made to `react-native-fbt` when I was writing the first draft of the playbook, and I do see that I have this method, `getTurboModule:`, implemented in the Fbt native module.

It's been a while since I have worked on this code, but based on the fact that further down below in this guide we do not explicitly implement a lookup method for this module in the `TurboModuleManagerDelegate`, it may be necessary to have this method implementation in place in the module.

I think it's okay to go ahead and include this block in the playbook.

Reply:
Included 👍